File: wachallenge/app/models/expense.rb
class Expense < ActiveRecord::Base
  belongs_to :report

  def self.upload_from_csv(file_param)
  	CSV.foreach(file_param.path, headers: true, skip_blanks: true, encoding:'windows-1256:utf-8') do |expense|
  		expense = expense.to_hash

  		Rails.logger.error expense.keys
  		Expense.create do |attrs|
  			attrs.date =  Date.strptime(expense["date"], "%m/%d/%Y")
  			attrs.category = expense["category"]
  			attrs.employee_name = expense["employee name"]
  			attrs.employee_address = expense["employee address"]
  			attrs.description = expense["expense description"]
  			attrs.pre_tax_amount = expense["pre-tax amount"]
  			attrs.tax_name = expense["tax name"]
  			attrs.tax_amount = expense["tax amount"]
  		end
  	end
  end
end

File: src/aspnetcoreapp/Models/Expense.cs
using System;

namespace WebApplication.Models
{
	public class Expense
	{
		//date,category,employee name,employee address,expense description,pre-tax amount,tax name,tax amount
		// could be broken into Employee, Tax etc
		public DateTime Date { get; set; }
		public string Category { get; set; }
		public string EmployeeName { get; set; }
		public string EmployeeAddress { get; set; }
		public string ExpenseDescription { get; set; }
		public double PreTaxAmount { get; set; }
		public string TaxName { get; set; }
		public double TaxAmount { get; set; }
	}
}
